src/api/routes/company.js

```diff
@@ -51,11 +55,48 @@ module.exports = (app) => {
         const computedOffset = parseInt(offset || 0, 10);
 
         try {
-            const { companies, totalDocCount } = await new CompanyService().findAll(computedLimit, computedOffset);
+            const { companies, totalDocCount } =
+                await new CompanyService().findAll(computedLimit, computedOffset, req.hasAdminPrivileges, req.hasAdminPrivileges);
             return res.json({ companies, totalDocCount });
         } catch (error) {
             return next(error);
         }
 
     });
+
+    router.put(
+        "/:companyId/block",
+        or([
+            authMiddleware.isGod,
+            authMiddleware.isAdmin
+        ],
+        { status_code: HTTPStatus.UNAUTHORIZED, error_code: ErrorTypes.FORBIDDEN, msg: ValidationReasons.INSUFFICIENT_PERMISSIONS }),
+        validators.block,
+        validators.canBlock,
+        async (req, res, _next) => {
+            const service = new CompanyService();
+            const company = await service.block(req.params.companyId, req.body.adminReason);
+            await service.sendCompanyBlockedNotification(req.params.companyId);
+            return res.json(company);
+        });
+
+    router.put(
+        "/:companyId/unblock",
+        or([
+            authMiddleware.isGod,
+            authMiddleware.isAdmin
+        ],
+        { status_code: HTTPStatus.UNAUTHORIZED, error_code: ErrorTypes.FORBIDDEN, msg: ValidationReasons.INSUFFICIENT_PERMISSIONS }),
+        validators.disable,
```

Comment:
Maybe this should match the route action?

```suggestion
        validators.unblock,
```